### DISCUSSION TITLE
Does the strongswan support SM2\SM3\SM4  encryption / Authentication Algorithm？

### DISCUSSION THREAD

**Discussion by yourounaida** [strongswan/strongswan#2323] [Q&A]
Does the strongswan support SM2\SM3\SM4  encryption / Authentication Algorithm？

**Comment by tobiasbrunner** [comment]
No. There are no IKEv2 identifiers yet anyway (see [this draft](https://datatracker.ietf.org/doc/html/draft-guo-ipsecme-ikev2-using-shangmi-00)). And plugins would also have to be updated (e.g. _openssl_) to support these algorithms.

**Comment by yourounaida** [comment]
Thanks for your reply. Wether IKEv1 support?




sviesvc
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "strongswan/strongswan"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月3日(星期三) 下午5:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [strongswan/strongswan] Does the strongswan support SM2\SM3\SM4  encryption / Authentication Algorithm？ (Discussion #2323)





 
No. There are no IKEv2 identifiers yet anyway (see this draft). And plugins would also have to be updated (e.g. openssl) to support these algorithms.

**Reply by tobiasbrunner** [reply]
What do you mean? There will never be support for these algorithms in IKEv1. It has been [officially deprecated](https://datatracker.ietf.org/doc/html/rfc9395) for over a year now, so no new algorithm identifiers will ever be defined for it.
